File: caesar.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <stdlib.h>

char caesar_shift(char c, int n);

int main(int argc, string argv[])
{
    // Error if wrong number of inputs
    if (argc != 2)
    {
        printf("Usage: caesar k\n");
        return 1;
    }
    // Prompt user for input
    printf("plaintext:");
    string s = get_string();
    // encrypt with caesar cypher
    printf("ciphertext:");
    for (int i = 0, n = strlen(s); i < n; i++)
    {
        printf("%c", caesar_shift(s[i], atoi(argv[1])));
    }
    printf("\n");
}

// maps letter and shift amount to new letter (mod alphabet), returns other characters without modification
char caesar_shift(char c, int n)
{
    if (c >= 'a' && c <= 'z')
    {
        return ((c + n - 'a') % 26) + 'a';
    } 
    else if (c >= 'A' && c <= 'Z')
    {
        return ((c + n - 'A') % 26) + 'A';
    }
    else
    {
        return c;
    }
}

File: caesar.py
import sys
import cs50

def main():
    #  Error if wrong number of inputs
    if len(sys.argv) != 2:
        print("Usage: caesar k")
        return 1
    
    # Prompt user for input
    s = cs50.get_string("plaintext:")
    
    # calls encrypts and returns results
    print("cyphertext:" + caesar_cypher(s, sys.argv[1]))
    
    # need this?
    return 0
    
def caesar_cypher(string, amt):
    t = ""
    for i in range(len(string)):
        t = t + caesar_shift(string[i], int(amt))
    return t
    
def caesar_shift(char, amt):
    if char >= 'a' and char <= 'z':
        return chr(((ord(char) + amt - 97) % 26) + 97)
    elif char >= 'A' and char <= 'Z':
        return chr(((ord(char) + amt - 65) % 26) + 65)
    else:
        return char
        
if __name__ == "__main__":
    main()